engine/access/state_stream/backend/backend_account_statuses_test.go

```diff
@@ -0,0 +1,582 @@
+	dssync "github.com/ipfs/go-datastore/sync"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/module/blobs"
+	"github.com/onflow/flow-go/module/execution"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data/cache"
+	"github.com/onflow/flow-go/module/mempool/herocache"
+	"github.com/onflow/flow-go/module/metrics"
+	protocolmock "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest/mocks"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/engine/access/index"
+	"github.com/onflow/flow-go/engine/access/state_stream"
+	"github.com/onflow/flow-go/engine/access/subscription"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+var testProtocolEventTypes = []flow.EventType{
+	state_stream.CoreEventAccountCreated,
+	state_stream.CoreEventAccountContractAdded,
+	state_stream.CoreEventAccountContractUpdated,
+}
+
+// Define the test type struct
+type testType struct {
+	name            string // Test case name
+	highestBackfill int    // Highest backfill index
+	startValue      interface{}
+	filters         state_stream.AccountStatusFilter // Event filters
+}
+
+// BackendAccountStatusesSuite is the test suite for AccountStatusesBackend.
+type BackendAccountStatusesSuite struct {
+	BackendExecutionDataSuite
+	accountCreatedAddress  flow.Address
+	accountContractAdded   flow.Address
+	accountContractUpdated flow.Address
+}
+
+func TestBackendAccountStatusesSuite(t *testing.T) {
+	suite.Run(t, new(BackendAccountStatusesSuite))
+}
+
+// generateProtocolMockEvents generates a set of mock events.
+func (s *BackendAccountStatusesSuite) generateProtocolMockEvents() flow.EventsList {
+	events := make([]flow.Event, 4)
+	events = append(events, unittest.EventFixture(testEventTypes[0], 0, 0, unittest.IdentifierFixture(), 0))
+
+	accountCreateEvent := generator.GenerateAccountCreateEvent(s.T(), s.accountCreatedAddress)
+	accountCreateEvent.TransactionIndex = 1
+	events = append(events, accountCreateEvent)
+
+	accountContractAdded := generator.GenerateAccountContractEvent(s.T(), "AccountContractAdded", s.accountContractAdded)
+	accountContractAdded.TransactionIndex = 2
+	events = append(events, accountContractAdded)
+
+	accountContractUpdated := generator.GenerateAccountContractEvent(s.T(), "AccountContractUpdated", s.accountContractUpdated)
+	accountContractUpdated.TransactionIndex = 3
+	events = append(events, accountContractUpdated)
+
+	return events
+}
+
+// SetupTest initializes the test suite.
+func (s *BackendAccountStatusesSuite) SetupTest() {
+	logger := unittest.Logger()
```

Comment:
instead of copying initialization from `BackendExecutionDataSuite` we can move initialization(the common part) in method and call it from the base struct.